ValveResourceFormat/IO/HammerMeshBuilder.cs

```diff
@@ -0,0 +1,1021 @@
+                            if (HalfEdges[possibleNext.twin].face == -1)
+                            {
+                                halfEdgeModifier.ChangeHalfEdgeNext(boundary, possibleNext.twin);
+                                halfEdgeModifier.ChangeHalfEdgePrev(HalfEdges[possibleNext.twin], boundaryIdx);
+                                break;
+                            }
+                        }
+                    }
+
+                }
+
+                if (totalPotentialNextBoundary == 1)
+                {
+                    if (potentialBoundaryWithAnInnerAsPrevIdx == -1)
+                    {
+                        var next = HalfEdges[HalfEdges[boundary.twin].prev].twin;
+                        halfEdgeModifier.ChangeHalfEdgeNext(boundary, next);
+                        halfEdgeModifier.ChangeHalfEdgePrev(HalfEdges[next], boundaryIdx);
+                    }
+                    else
+                    {
+                        halfEdgeModifier.ChangeHalfEdgeNext(boundary, potentialBoundaryWithAnInnerAsPrevIdx);
+                        halfEdgeModifier.ChangeHalfEdgePrev(HalfEdges[potentialBoundaryWithAnInnerAsPrevIdx], boundaryIdx);
+                    }
+                }
+            }
+
+            Faces.Add(face);
+        }
+
+        internal class HalfEdgeMeshModifier
+        {
+            private List<HalfEdgeModification> HalfEdgeModifications = new();
+            private List<VertsToEdgeDictModification> VertsToEdgeDictModifications = new();
+            private List<HalfEdgesModification> HalfEdgesModifications = new();
+            private List<VertexModification> VertexModifications = new();
+
+            public HammerMeshBuilder.Face Face;
+
+            private HammerMeshBuilder builderRef;
+
+            public HalfEdgeMeshModifier(HammerMeshBuilder c_builderRef)
+            {
+                builderRef = c_builderRef;
+            }
+
+            // remember half edge states
+            internal class HalfEdgeModification
+            {
+                public HammerMeshBuilder.HalfEdge he;
+                public string propertyName;
+                public int property;
+
+                public HalfEdgeModification(HammerMeshBuilder.HalfEdge c_he, string c_propertyName, int c_property)
+                {
+                    he = c_he;
+                    propertyName = c_propertyName;
+                    property = c_property;
+                }
+            }
+            internal void RememberHalfEdgeModification(HammerMeshBuilder.HalfEdge he, string propertyName, int property)
+            {
+                HalfEdgeModifications.Add(new HalfEdgeModification(he, propertyName, property));
+            }
+
+            public void ChangeHalfEdgeFace(HammerMeshBuilder.HalfEdge he, int face)
+            {
+                if (he.face == face)
+                {
+                    return;
+                }
+
+                RememberHalfEdgeModification(he, "face", he.face);
```

Comment:
```suggestion
                RememberHalfEdgeModification(he, nameof(HalfEdge.face), he.face);
```

same for the rest

Reply:
These probably shouldn't be strings at all.

Reply:
Yeah they should be an enum